app/scripts/lib/safe-methods.json

```diff
@@ -0,0 +1,48 @@
+[
+  "web3_sha3",
+  "net_listening",
+  "net_peerCount",
+  "net_version",
+  "eth_blockNumber",
+  "eth_call",
+  "eth_coinbase",
+  "eth_estimateGas",
+  "eth_gasPrice",
+  "eth_getBalance",
+  "eth_getBlockByHash",
+  "eth_getBlockByNumber",
+  "eth_getBlockTransactionCountByHash",
+  "eth_getBlockTransactionCountByNumber",
+  "eth_getCode",
+  "eth_getFilterChanges",
+  "eth_getFilterLogs",
+  "eth_getLogs",
+  "eth_getStorageAt",
+  "eth_getTransactionByBlockHashAndIndex",
+  "eth_getTransactionByBlockNumberAndIndex",
+  "eth_getTransactionByHash",
+  "eth_getTransactionCount",
+  "eth_getTransactionReceipt",
+  "eth_getUncleByBlockHashAndIndex",
+  "eth_getUncleByBlockNumberAndIndex",
+  "eth_getUncleCountByBlockHash",
+  "eth_getUncleCountByBlockNumber",
+  "eth_getWork",
+  "eth_hashrate",
+  "eth_mining",
+  "eth_newBlockFilter",
+  "eth_newFilter",
+  "eth_newPendingTransactionFilter",
+  "eth_protocolVersion",
+  "eth_sendRawTransaction",
+  "eth_sendTransaction",
```

Comment:
an unauthorized dapp may be able to detect if you control a known address by trying suggesting txs/signatures with various from addresses. Addresses you dont control will immediately fail but allowed addresses will wait for user confirmation.